cmd/status/cmdstatus.go

```diff
@@ -374,3 +372,42 @@ func (ir *InventoryLoader) GetInvInfo(cmd *cobra.Command, args []string) (invent
 	inv := inventory.WrapInventoryInfoObj(invObj)
 	return inv, nil
 }
+
+func (r *Runner) listClusterInventoryObjs(ctx context.Context) (map[string]object.ObjMetadataSet, error) {
```

Comment:
I was hoping we could hide the listing of inventory objects inside the inventory client, making it an implementation detail the caller doesn't know about. But maybe that needs to be a subsequent change?

The `map[string]object.ObjMetadataSet` here looks like it's just being looped through to get a full list of "Identifiers" for the `PrintData` built by `listInvFromCluster`. That flattening could just be done inside the inventory client.

That would require changing the `PrintData` to remove the `InvNameMap` and then updating `printStatusEvent`  in the json & event printers. Then instead of printing the actual inventory object name, we print the ID that might be the name or label.

I don't see anywhere in kpt that uses the "inventory-name" field from the json printer. So it probably wouldn't break kpt itself. That said, I have no idea if anyone uses the json output from kpt live status for integration purposes. Even so, the status event output doesn't have a field for "inventory-namespace", so it would be hard for a caller to depend on the name to do their own GET or anything.

Reply:
This method is only being used by the `status` CLI command, so it doesn't need to be supported by all implementations of the interface. The CLI is statically built from this repo and only supports the ConfigMap type, so the implementation only needs to support ConfigMap.

I'm thinking of this method as an edge case that doesn't need to be implemented by all instances of the Client/Inventory interface. This will simplify the further refactoring of the inventory client interface and introductions of new types of clients

Reply:
`kpt live status` actually uses the status.Runner built for `kapply status`. So it indirectly uses this code to build the `PrintData`, which is used both for printing and passed to the status poller/watcher.

My thinking is that this all needs refactoring to avoid exposing the inventory names and only expose a single inventory id.

If we do that, then we can push the listing down into the `Client.GetClusterObjs` or similar getter that returns the list of managed objects.

Additionally, we can replace `Client.GetClusterObjs` with something simpler like `Client.GetObjectIDs(id) ObjMetadataSet` that just returns the list of IDs from the inventory spec. Then a utility func can be used to get those objects from either the cluster or some other in-memory object cache. We don't need the inventory client to hide the whole cluster, just the inventory management.

Reply:
Or maybe `inventory.Client.GetSpec(id) actuation.InventorySpec`.

I don't know if we need a `inventory.Client.GetStatus(id) actuation.InventoryStatus`, but that's the other logical getter.

We just don't want to `GetInventory(id) actuation.Inventory`, because that includes TypeMeta & ObjectMeta which requires the inventory to be a single object.

Reply:
If you want to do that in a subsequent PR, LMK.

I think overall removing these methods from the Client interface is a good start. But we'll need to move this list code again later.

Reply:
> My thinking is that this all needs refactoring to avoid exposing the inventory names and only expose a single inventory id. If we do that, then we can push the listing down into the Client.GetClusterObjs or similar getter that returns the list of managed objects.

Note that this method returns _all inventories_ in the cluster, as opposed to _managed objects for a given inventory_ in the cluster. It seems a bit strange to me to have as part of the interface, as it appears to be specific to the use case of the status command (unused by the applier).

I didn't realize that kpt imports the status command to construct the `status.Runner` for ResourceGroup. That makes this a bit more sticky, so we might need to keep the listing method attached to the inventory client interface in that case. This means we'll need to decide how the listing interface should look for sharded inventories. I assume we will want each value in the key/value pair to represent a joined inventory, as opposed to having a separate key/value pair for each inventory shard.